zio-cli/shared/src/main/scala/zio/cli/PrimType.scala

```diff
@@ -36,20 +40,101 @@ object PrimType {
-  case object Integer                                        extends PrimType[BigInt]
-  case object Instant                                        extends PrimType[JInstant]
-  case object LocalDate                                      extends PrimType[JLocalDate]
-  case object LocalDateTime                                  extends PrimType[JLocalDateTime]
-  case object LocalTime                                      extends PrimType[JLocalTime]
-  case object MonthDay                                       extends PrimType[JMonthDay]
-  case object OffsetDateTime                                 extends PrimType[JOffsetDateTime]
-  case object OffsetTime                                     extends PrimType[JOffsetTime]
-  case object Period                                         extends PrimType[JPeriod]
-  case object Year                                           extends PrimType[JYear]
-  case object YearMonth                                      extends PrimType[JYearMonth]
-  case object ZonedDateTime                                  extends PrimType[JZonedDateTime]
-  case object ZoneId                                         extends PrimType[JZoneId]
-  case object ZoneOffset                                     extends PrimType[JZoneOffset]
+  {
+    import PathType._
+    def validate(value: String): IO[String, JPath] = 
+      for {
+        p       <- IO.effect(JPaths.get(value)).catchAll(_ => IO.fail(s"Couldn't recognize '$value' as valid path."))
+        _       <- exists(p) >>= refine("Expected path '$value' to exist.")
+        _       <- pathType match {
+                      case Anything => IO.unit
+                      case File => isRegularFile(p) >>= refine(s"Expected path '$value' to be a regular file.")
+                      case Directory => isDirectory(p) >>= refine(s"Expected path '$value' to be a directory.")
+                    }
+      } yield p
+      
+    private def exists(path: JPath) = IO.effect(JFiles.exists(path)) orElse IO.succeed(false)
+
+    private def isDirectory(path: JPath) = IO.effect(JFiles.isDirectory(path)) orElse IO.succeed(false)
+
+    private def isRegularFile(path: JPath) = IO.effect(JFiles.isDirectory(path)) orElse IO.succeed(false)
+
+    private def refine(message: String) (f: Boolean) = if (f) IO.unit else IO.fail(message)
+  }
+  
+  case object Text extends PrimType[String] {
+    def validate(value: String): IO[String, String] = attempt(value, _ => value, "text")
+  }
+  
+  case object Decimal extends PrimType[BigDecimal] {
+    def validate(value: String): IO[String, BigDecimal] = attempt(value, BigDecimal(_), "decimal")
+  }
+  
+  case object Integer extends PrimType[BigInt] {
+    def validate(value: String): IO[String, BigInt] = attempt(value, BigInt(_), "integer")
+  }
+
+  case object Boolean extends PrimType[Boolean] {
+    def validate(value: String): IO[String, Boolean] = value match {
```

Comment:
I'd tend toward generosity here, so do `value.trim.toLowerCase`, maybe?

```suggestion
    def validate(value: String): IO[String, Boolean] = value.trim.toLowerCase match {
```

Reply:
Agree, adjusted.